corehq/messaging/management/commands/export_conditional_alerts.py

```diff
@@ -0,0 +1,209 @@
+            use_utc_as_default_timezone=schedule.use_utc_as_default_timezone,
+            user_data_filter=copy.deepcopy(schedule.user_data_filter),
+            stop_date_case_property_name=schedule.stop_date_case_property_name,
+        )
+
+    def get_json_timed_schedule(self, schedule):
+        if schedule.ui_type != Schedule.UI_TYPE_DAILY:
+            raise CommandError("Expected simple daily schedule")
+
+        json_schedule = SimpleSMSDailyScheduleWithTime(
+            total_iterations=schedule.total_iterations,
+            start_offset=schedule.start_offset,
+            start_day_of_week=schedule.start_day_of_week,
+            repeat_every=schedule.repeat_every,
+            extra_options=self.get_json_scheduling_options(schedule),
+        )
+
+        event = schedule.memoized_events[0]
+        if not isinstance(event, TimedEvent):
+            raise CommandError("Expected TimedEvent")
+
+        json_schedule.time = event.time
+
+        content = event.content
+        if not isinstance(content, SMSContent):
+            raise CommandError("Expected SMSContent")
+
+        json_schedule.message = copy.deepcopy(content.message)
+
+        return json_schedule
+
+    def get_json_alert_schedule(self, schedule):
+        if schedule.ui_type != Schedule.UI_TYPE_IMMEDIATE:
+            raise CommandError("Expected simple immediate schedule")
+
+        json_schedule = SimpleSMSAlertSchedule(
+            extra_options=self.get_json_scheduling_options(schedule),
+        )
+
+        event = schedule.memoized_events[0]
+        content = event.content
+        if not isinstance(content, SMSContent):
+            raise CommandError("Expected SMSContent")
+
+        json_schedule.message = copy.deepcopy(content.message)
+
+        return json_schedule
+
+    def handle(self, domain, **options):
+        result = []
+        for rule in AutomaticUpdateRule.by_domain(
+            domain,
+            AutomaticUpdateRule.WORKFLOW_SCHEDULING,
+            active_only=False,
+        ):
+            json_rule = self.get_json_rule(rule)
+
+            action = rule.memoized_actions[0].definition
+            if action.schedule.location_type_filter:
+                raise CommandError("Expected location_type_filter to be empty for rule %s" % rule.pk)
+
+            if isinstance(action.schedule, TimedSchedule):
+                json_schedule = self.get_json_timed_schedule(action.schedule)
+            elif isinstance(action.schedule, AlertSchedule):
+                json_schedule = self.get_json_alert_schedule(action.schedule)
+            else:
+                raise CommandError("Unexpected Schedule type for rule %s" % rule.pk)
+
+            result.append(json.dumps({
+                'rule': json_rule.to_json(),
+                'schedule': json_schedule.to_json(),
+            }))
+
+        with open('conditional_alerts_for_%s.txt' % domain, 'wb') as f:
+            for line in result:
+                f.write(line)
```

Comment:
I think this will work on Python 2, but not Python 3 because on Python 2 `json.dumps` returns bytes and on Python 3 it returns text (unicode). See https://github.com/dimagi/langcodes/pull/19#discussion_r189239788 for some possibilities of how to make it work on both.

Reply:
updated for py3 compatibility here: https://github.com/dimagi/commcare-hq/pull/21366/commits/606b606f75b6e19af4dc24038fece9713af6318d